x-pack/plugins/lists/public/exceptions/hooks/use_api.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useMemo } from 'react';
+
+import * as Api from '../api';
+import { HttpStart } from '../../../../../../src/core/public';
+import { ExceptionListItemSchema, ExceptionListSchema } from '../../../common/schemas';
+import { ApiCallMemoProps } from '../types';
+
+export interface ExceptionsApi {
+  deleteExceptionItem: (arg: ApiCallMemoProps) => Promise<void>;
+  deleteExceptionList: (arg: ApiCallMemoProps) => Promise<void>;
+  getExceptionItem: (
+    arg: ApiCallMemoProps & { onSuccess: (arg: ExceptionListItemSchema) => void }
+  ) => Promise<void>;
+  getExceptionList: (
+    arg: ApiCallMemoProps & { onSuccess: (arg: ExceptionListSchema) => void }
+  ) => Promise<void>;
+}
```

Comment:
Nice way of exposing the list API's! Makes for a clean separation of concerns between usage in hooks, and exposing the plugin's API as a whole 🙂

Reply:
Yea, the other way felt strange and saw this pattern elsewhere.